xclim/indices/_conversion.py

```diff
@@ -695,3 +697,143 @@ def wind_chill_index(
+
+    Returns
+    -------
+    xarray.DataArray
+        Potential evapotranspiration (daily or monthly).
+
+    Notes
+    -----
+    Available methods are:
+
+    - "baierrobertson65" or "BR65", based on [baierrobertson65]_.
+    - "hargreaves85" or "HG85", based on [hargreaves85]_.
+    - "thornthwaite48" or "TW48", based on [thornthwaite48]_.
+
+    References
+    ----------
+    .. [baierrobertson65] Baier, W., & Robertson, G. W. (1965). Estimation of latent evaporation from simple weather observations. Canadian journal of plant science, 45(3), 276-284.
+    .. [hargreaves85] Hargreaves, G. H., & Samani, Z. A. (1985). Reference crop evapotranspiration from temperature. Applied engineering in agriculture, 1(2), 96-99.
+    .. [thornthwaite48] Thornthwaite, C. W. (1948). An approach toward a rational classification of climate. Geographical review, 38(1), 55-94.
+    """
+
+    if method in ["baierrobertson65", "BR65"]:
+        tasmin = convert_units_to(tasmin, "degF")
+        tasmax = convert_units_to(tasmax, "degF")
+
+        latr = (tasmin.lat * np.pi) / 180
+        gsc = 0.082  # MJ/m2/min
+
+        # julian day fraction
+        jd_frac = (datetime_to_decimal_year(tasmin.time) % 1) * 2 * np.pi
+
+        ds = 0.409 * np.sin(jd_frac - 1.39)
+        dr = 1 + 0.033 * np.cos(jd_frac)
+        omega = np.arccos(-np.tan(latr) * np.tan(ds))
+        re = (
+            (24 * 60 / np.pi)
+            * gsc
+            * dr
+            * (
+                omega * np.sin(latr) * np.sin(ds)
+                + np.cos(latr) * np.cos(ds) * np.sin(omega)
+            )
+        )  # MJ/m2/day
+        re = re / 4.1864e-2  # cal/cm2/day
+
+        # Baier et Robertson(1965) formula
+        out = 0.094 * (
+            -87.03 + 0.928 * tasmax + 0.933 * (tasmax - tasmin) + 0.0486 * re
+        )
+        out = out.clip(0)
+        out.attrs["units"] = "mm/day"
+
+    elif method in ["hargreaves85", "HG85"]:
+        tasmin = convert_units_to(tasmin, "degC")
+        tasmax = convert_units_to(tasmax, "degC")
+        tas = convert_units_to(tas, "degC")
+
+        latr = (tasmin.lat * np.pi) / 180
+        gsc = 0.082  # MJ/m2/min
+        lv = 2.5  # MJ/kg
+
+        # julian day fraction
+        jd_frac = (datetime_to_decimal_year(tasmin.time) % 1) * 2 * np.pi
+
+        ds = 0.409 * np.sin(jd_frac - 1.39)
+        dr = 1 + 0.033 * np.cos(jd_frac)
+        omega = np.arccos(-np.tan(latr) * np.tan(ds))
+        ra = (
+            (24 * 60 / np.pi)
+            * gsc
+            * dr
+            * (
+                omega * np.sin(latr) * np.sin(ds)
+                + np.cos(latr) * np.cos(ds) * np.sin(omega)
+            )
+        )  # MJ/m2/day
+
+        # Hargreaves and Samani(1985) formula
+        out = (0.0023 * ra * (tas + 17.8) * (tasmax - tasmin) ** 0.5) / lv
+        out = out.clip(0)
+        out.attrs["units"] = "mm/day"
+
+    elif method in ["thornthwaite48", "TW48"]:
+        tas = convert_units_to(tas, "degC")
```

Comment:
I think one of the big advantages of thonthwaite is to be able to calculate it on data that is already monthly.   But the following code doesn't give the same result. 
```
pet_tw48 = atmos.potential_evapotranspiration(tn, tx, tm, method="TW48")

 pet_tw48C = atmos.potential_evapotranspiration(tn, tx,
                                                        tm.resample(time='MS').mean(dim='time', keep_attrs=True),
                                                        method="TW48")
```

I think the main issue is wrt to the daylength calc. If input data is already monthly the daylength calculation would need to rebuild it's own complete daily `time` variable  instead of directly using `tas.time` (line 810)

`jd_frac = (datetime_to_decimal_year(tas.time) % 1) * 2 * np.pi`